precommit warns on main + instructions for fix (#26171)
matplotlib/matplotlib: doc/devel/development_workflow.rst

    git branch my-new-feature upstream/main
    git checkout my-new-feature

If you started making changes on your local ``main`` branch, you can convert the
branch to a feature branch by renaming it::

>    git branch -m <newname>
> 
> 
> Generally, you will want to keep your feature branches on your public GitHub

Reviewer: ```suggestion
   git branch -m <newname>

Generally, you will want to keep your feature branches on your public GitHub
```

From now on git will know that ``my-new-feature`` is related to the
``my-new-feature`` branch in the GitHub repo.

> If you first opened the pull request on ``main`` and then converted it to a
> feature branch, you will need to close the original pull request and open a new
> one for the renamed branch. See
> `Github: working with branches <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.

Reviewer 2: ```suggestion
If you first opened the pull request from your ``main`` branch and then converted it to a
feature branch, you will need to close the original pull request and open a new
one from the renamed branch. See
`Github: working with branches <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.
```

> `Github: working with branches <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.
> 
> 
> .. _edit-flow:

Reviewer: ```suggestion
`GitHub: working with branches
<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.

.. _edit-flow:
```
